### PR TITLE
feat: Add payments tests against CSV values (SC-3899 + SC-3900 + SC-3901)

### PR DESCRIPTION
**Pull request by lucas-manuel**
# Description

Add payments tests for interest only, partially amortized, and fully amortized loans against the sheet provided by the business team: https://docs.google.com/spreadsheets/d/1C_zhjPs_s_IBnDYBOdrfF2jd59vLAEXumN49YbmTJns/edit#gid=1232787594

All payments are made on time.

Rates have been updated to use `1e18` precision instead of basis points * 100

# Integrations Checklist

- [ ] Have any function signatures changed? If yes, outline below.
- [ ] Have any features changed or been added? If yes, outline below.
- [ ] Have any events changed or been added? If yes, outline below.
- [ ] Has all documentation been updated?

# Changelog

### CODE REVIEW

File: contracts/test/Payments.t.sol [hunk 1]
```diff
@@ -0,0 +1,552 @@
...
+            assertIgnoringDecimals(loan.drawableFunds(),  0,                     13);  // No extra funds left in contract TODO: Try assertEq
+            assertIgnoringDecimals(loan.principal(),      principalRemaining[i], 13);  // Principal decreasing in accordance with provided values
+
+            assertEq(loan.paymentsRemaining(),  6 - (i + 1));                               // Payments remaining increases
+            assertEq(loan.nextPaymentDueDate(), start + loan.paymentInterval() * (i + 2));  // Payment due date increases
+        }
+
+        assertEq(loan.principal(), 0);
+
+        // TODO: Add final balance variable
```
Comment (lucas-manuel) [inline contracts/test/Payments.t.sol:133]: Conflicted on whether this is really necessary, lmk what you guys think

File: contracts/test/Payments.t.sol [hunk 2]
```diff
@@ -0,0 +1,552 @@
...
+        /*** Loan Terms:	
+        /*** Amount	                   1,000,000
+        /*** Ending Pricipal           0
+        /*** Interest Rate	           10%
+        /*** Payment Interval (days)   30
+        /*** Term	                   180
+        /*** Number of payments        6
+        /*** Interest Rate per period  0.82%
+        /*** Payment Amount per period 171,494
+        /****************************************/
```
Comment (lucas-manuel) [inline contracts/test/Payments.t.sol:152]: Just noticed these are misaligned, will push a fix soon

File: contracts/test/Payments.t.sol (hunk 1 above)
Comment (deluca-mike) [reply contracts/test/Payments.t.sol:133]: I would do it just for completeness.

File: contracts/test/Payments.t.sol (hunk 2 above)
Comment (deluca-mike) [reply contracts/test/Payments.t.sol:152]: Make sure to "Remove Trailing Whitespaces" in VS Code.

File: contracts/test/Payments.t.sol [hunk 3]
```diff
@@ -0,0 +1,552 @@
...
+        ];
+
+        MapleLoan loan = createLoanFundAndDrawdown(assets, parameters, requests);
+
+        paymentsTest(loan, parameters, requests, principalPortions, interestPortions, principalRemaining, totals);
+    }
+
+    function test_payments_fullyAmortized_case2() external {
+                               
+        /****************************************/
```
Comment (deluca-mike) [inline contracts/test/Payments.t.sol:210]:
```suggestion
    function test_payments_fullyAmortized_case2() external {
        /****************************************/
```
and the rest as well.

**Review by JGcarv — CHANGES_REQUESTED** [review]

**Review by deluca-mike — APPROVED** [review]

**Pull request merged**